src/main/java/com/laytonsmith/PureUtilities/Point3D.java

```diff
@@ -0,0 +1,127 @@
+package com.laytonsmith.PureUtilities;
+
+/**
+ * Represents a point in 3D space.
+ *
+ */
+public class Point3D {
+
+	/**
+	 * A Point3D located at [0, 0, 0] (zero)
+	 */
+	public static Point3D ZERO = new Point3D();
+
+    protected double x;
```

Comment:
Since these are immutable, you should make these properties final, so they can't be changed.